Podverse/PVDeleter.swift

```diff
@@ -10,166 +10,83 @@ import Foundation
-                //                    PVMediaPlayer.sharedInstance.nowPlayingEpisode = nil
-                //                    NSUserDefaults.standardUserDefaults().removeObjectForKey(Constants.kLastPlayingEpisodeURL)
-                //
-                //                    NSNotificationCenter.defaultCenter().postNotificationName(Constants.kPlayerHasNoItem, object: nil)
-                //                }
-                //            }
             }
             
-            // Delete the episode from CoreData and the disk, and update the UI
             if let fileName = episode.fileName {
-                PVDeleter
-                    .deleteEpisodeFromDiskWithName(fileName: fileName)
-                episode.fileName = nil
+                PVDeleter.deleteEpisodeFromDiskWithName(fileName: fileName)
             }
             
-            // If the episode or a clip from the episode is currently a playlistItem in a local playlist, then do not delete the episode item from Core Data
-            if checkIfEpisodeShouldBeRemoved(episode) == true {
-                CoreDataHelper.deleteItemFromCoreData(deleteObjectID: episode.objectID, moc: moc)
-            }
+            CoreDataHelper.deleteItemFromCoreData(deleteObjectID: episode.objectID, moc: moc)
             
             moc.saveData(nil)
             
-        }
-        
-    }
-    
-    static func checkIfEpisodeShouldBeRemoved(_ episode: Episode) -> Bool {
-        let moc = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
-        moc.parent = CoreDataHelper.shared.managedObjectContext
-        
-        var alsoDelete = true
-        
-        if episode.podcast.isSubscribed == true || episode.podcast.isFollowed == true {
-            alsoDelete = false
-            return alsoDelete
-        }
-        
-        if let allPlaylists = CoreDataHelper.fetchEntities(className: "Playlist", predicate: nil, moc: moc) as? [Playlist] {
-            outerLoop: for playlist in allPlaylists {
-                for item in playlist.allItems {
-                    if let episode = item as? Episode {
-                        for ep in episode.podcast.episodes {
-                            if ep.objectID == episode.objectID {
-                                alsoDelete = false
-                                break outerLoop
-                            }
-                        }
-                    }
-//                    else if let clip = item as? Clip {
-//                        for ep in episode.podcast.episodes {
-////                            for cl in ep.clips {
-////                                if clip.objectID == cl.objectID {
-////                                    alsoDelete = false
-////                                    break outerLoop
-////                                }
-////                            }
-//                        }
-//                    }
```

Comment:
Sure. Decrement badge is UI and the episodeDeleted implementation could potentially do UI stuff so its fine by me